src/value.rs

```diff
@@ -927,6 +928,24 @@ mod tests {
             assert_eq!(Value::zero(&ty), expected_default_value);
         }
     }
+
+    #[test]
+    fn compact_round_trip() {
+        let v = Value::u64(0xff00_00ff_ff00_00ff);
+        let (bits, _) = v.iter_compact().collect_bits();
+        let mut iter = BitIter::new(bits.into_iter());
+        let new_v = Value::from_compact_bits(&mut iter, &v.ty).unwrap();
+        assert_eq!(v, new_v);
+    }
+
+    #[test]
+    fn padded_round_trip() {
+        let v = Value::u64(0xff00_00ff_ff00_00ff);
+        let (bits, _) = v.iter_padded().collect_bits();
+        let mut iter = BitIter::new(bits.into_iter());
+        let new_v = Value::from_padded_bits(&mut iter, &v.ty).unwrap();
+        assert_eq!(v, new_v);
+    }
```

Comment:
790b38a2abe7883c393ab8be1e14906feb5c5112: Value encoding roundtrips are candidates for future fuzz tests